code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -361,97 +318,188 @@
 
 /obj/machinery/kitchen_machine/proc/start()
 	visible_message("<span class='notice'>\The [src] turns on.</span>", "<span class='notice'>You hear \a [src].</span>")
+	if(soundloop)
+		soundloop.start()
+	else
+		playsound(loc, 'sound/machines/click.ogg', 50, 1)
 	operating = TRUE
-	icon_state = on_icon
-	updateUsrDialog()
+	update_icon(UPDATE_ICON_STATE)
+	SStgui.update_uis(src)
+
+/obj/machinery/kitchen_machine/proc/finish_sound()
+	if(soundloop)
+		soundloop.stop()
+	else
+		playsound(loc, 'sound/machines/ding.ogg', 50, 1)
 
 /obj/machinery/kitchen_machine/proc/abort()
 	operating = FALSE // Turn it off again aferwards
-	icon_state = off_icon
-	updateUsrDialog()
+	update_icon(UPDATE_ICON_STATE)
+	SStgui.update_uis(src)
 
 /obj/machinery/kitchen_machine/proc/stop()
-	playsound(loc, 'sound/machines/ding.ogg', 50, 1)
+	finish_sound()
 	operating = FALSE // Turn it off again aferwards
-	icon_state = off_icon
-	updateUsrDialog()
-
-/obj/machinery/kitchen_machine/proc/dispose()
-	for(var/obj/O in contents)
-		O.forceMove(loc)
-	if(reagents.total_volume)
-		dirty++
-	reagents.clear_reagents()
-	to_chat(usr, "<span class='notice'>You dispose of \the [src]'s contents.</span>")
-	updateUsrDialog()
+	update_icon(UPDATE_ICON_STATE)
+	SStgui.update_uis(src)
 
 /obj/machinery/kitchen_machine/proc/muck_start()
 	playsound(loc, 'sound/effects/splat.ogg', 50, 1) // Play a splat sound
-	icon_state = dirty_icon // Make it look dirty!!
+	dirty = 100 // Make it dirty so it can't be used util cleaned
+	update_icon(UPDATE_ICON_STATE)
 
 /obj/machinery/kitchen_machine/proc/muck_finish()
-	playsound(loc, 'sound/machines/ding.ogg', 50, 1)
 	visible_message("<span class='alert'>\The [src] gets covered in muck!</span>")
-	dirty = 100 // Make it dirty so it can't be used util cleaned
-	flags = null //So you can't add condiments
-	icon_state = dirty_icon // Make it look dirty too
-	operating = FALSE // Turn it off again aferwards
-	updateUsrDialog()
-
-/obj/machinery/kitchen_machine/proc/broke()
-	do_sparks(2, 1, src)
-	icon_state = broken_icon // Make it look all busted up and shit
-	visible_message("<span class='alert'>[src] breaks!</span>") //Let them know they're stupid
-	broken = 2 // Make it broken so it can't be used util fixed
 	flags = null //So you can't add condiments
-	operating = FALSE // Turn it off again aferwards
-	updateUsrDialog()
+	stop()
 
 /obj/machinery/kitchen_machine/proc/fail()
 	var/amount = 0
 	for(var/obj/item/mixing_bowl/mb in contents)	//fail and remove any mixing bowls present before making the burned mess from the machine itself (to avoid them being destroyed as part of the failure)
 		mb.fail(src)
 		mb.forceMove(get_turf(src))
 	for(var/obj/O in contents)
```

Comment:
it'd be nice to see all these loops through contents instead just use a list that tracks ingredients inside the machine

Reply:
I tried looking into this and getting it to work, and it seems more complicated than helpful, it would basically just be contents + reagents, which would make it more confusing